app/main.py

```diff
@@ -1,8 +1,27 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
+
+    def add_husband_wife_if_merried(self, person: dict) -> None:
+        if person.get("wife") is not None:
+            self.wife = Person.people.get(person.get("wife"))
+
+        if person.get("husband") is not None:
+            self.husband = Person.people.get(person.get("husband"))
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    for person in people:
```

Comment:
two identical cycles, maybe they can be combined

Reply:
they are not identical, first is to initialize instances of a class Person, second to link wife/husband reference to a instance of a class if he/she has wife/husband.

Reply:
Yes, I see, if do that in one cycle, can be error